### ISSUE TITLE
[FEATURE REQUEST] Can ruby-build check for dependencies before starting the build process?

### ISSUE THREAD

**Issue by rlue** [rbenv/ruby-build#1113]
I'm on a fresh install of openSUSE Tumbleweed (13.3), and was having trouble building any version of Ruby because I didn't have `libopenssl-devel` or `readline-devel`. 

(I also didn't have `libffi-devel`, but I'm not 100% sure that I really needed it -- I installed it out of desperation, based on what I'd read in #731, and not based on the messages from my error logs.)

I got it to work in the end, but it took me a long time to notice the relevant message in the log (though I admit it could not have been more obvious):

```
The Ruby openssl extension was not compiled.
The Ruby readline extension was not compiled.
ERROR: Ruby install aborted due to missing extensions
Try running `apt-get install -y libssl-dev libreadline-dev` to fetch missing dependencies.
```

I think what threw me off was the 10,000+ line log file and the five-minute-long build process. Is there any way ruby-build could incorporate a dependency-checker that runs _before_ the build process starts, rather than running into errors after a five-minute compile and throwing the relevant messages in at the end of the logs?

**Comment by giorgian** [comment]
Please please please.
Solving this would solve quite a number of other issues on this project.

**Comment by eregon** [comment]
Unfortunately this is not feasible given there are various dependency per version and per Ruby implementation.
We need to rely on those Ruby implementations doing early checks.

FWIW there was a e.g. llvm@4 check for truffleruby but it turned out to be more annoying than helpful, when the dependency was no longer needed and that dependency was actually removed from Homebrew.
